modules/nw-ingress-creating-a-route-via-an-ingress.adoc

```diff
@@ -0,0 +1,99 @@
+// Module included in the following assemblies:
+//
+// * ingress/routes.adoc
+
+[id="nw-ingress-creating-a-route-via-an-ingress_{context}"]
+= Creating a Route via an Ingress
+
+Some ecosystem components have an integration with Ingress resources but not with
+Route resources. To cover for this case, Openshift will automatically create
+managed Route object(s) when an Ingress is created. These Route objects are deleted
+upon deletion of the corresponding Ingress.
+
+.Procedure
+
+Define an Ingress object:
++
+.YAML Definition of an Ingress
+[source,yaml]
+----
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: frontend
+  annotations:
+    route.openshift.io/termination: "edge" <1>
+spec:
+  rules:
+  - host: www.example.com
+    http:
+      paths:
+      - backend:
+          serviceName: frontend
+          servicePort: 443
+  tls:
+  - hosts:
+    - www.example.com
+    secretName: example-com-tls-certificate
+----
++
+
+<1> The `route.openshift.io/termination` annotation can be used to configure the `termination` of the `Route`
+as `Ingress` has no property for this. The accepted values are `edge`, `passthrough` and `reencrypt`. All
+other values will be silently ignored. When unset, `edge` will be used.
+
+Apply it to your cluster:
++
+----
+$ oc apply -f ingress.yaml
+----
++
+
+List your routes:
++
+----
+$ oc get routes
+----
++
+
+The result will include an autogenerated route whose name starts with `frontend-`:
++
+----
+NAME             HOST/PORT         PATH    SERVICES    PORT    TERMINATION          WILDCARD
+frontend-gnztq   www.example.com           frontend    443     reencrypt/Redirect   None
+----
++
+
+If you inspect this Route, it will look like this:
++
+.YAML Definition of an autogenerated Route
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: frontend-gnztq
+  ownerReferences:
+  - apiVersion: extensions/v1beta1
+    controller: true
+    kind: Ingress
+    name: frontend
+    uid: 4e6c59cc-704d-4f44-b390-617d879033b6
+spec:
+  host: www.example.com
+  to:
+    kind: Service
+    name: frontend
+  tls:
+    certificate: |
+      -----BEGIN CERTIFICATE-----
+      [...]
+      -----END CERTIFICATE-----
+    insecureEdgeTerminationPolicy: Redirect
+    key: |
+      -----BEGIN RSA PRIVATE KEY-----
+      [...]
+      -----END RSA PRIVATE KEY-----
+    termination: edge
```

Comment:
Here the example route yaml matches the example ingress yaml but not the example `oc get routes` output.  If you change the ingress yaml (as opposed to changing the `oc get routes` example output), remember to change the route yaml.